generators/csharp/codegen/src/ast/ClassInstantiation.ts

```diff
@@ -45,28 +45,44 @@ export class ClassInstantiation extends AstNode {
             writer.write("(");
         }
 
-        writer.newLine();
-        writer.indent();
+        if (this.arguments_.length === 0) {
+            this.writeEnd({ writer, hasNamedArguments });
+            return;
+        }
+
+        if (this.multiline) {
+            writer.newLine();
+        }
 
+        writer.indent();
         this.arguments_.forEach((argument, idx) => {
             if (isNamedArgument(argument)) {
-                writer.write(`${argument.name} = `);
+                writer.write(`${argument.name}`);
+                if (this.forceUseConstructor) {
+                    writer.write(": ");
+                } else {
+                    writer.write(" = ");
+                }
```

Comment:
Ideally, we should disambiguate arguments from properties.
Arguments can be positional or named, but they are not properties.
Then any properties passed to the class instantiation should be used as property initializer.
This is fine for now tho, but this `this.forceUseConstructor` doesn't really make sense and is a workaround addressing the ambiguity.

Reply:
Good call, I agree that the `forceUseConstructor` use case felt a little off. I ended up keeping things as-is for now, but I'm in favor of coming back to this to clean it up later.